### PR TITLE
Task #9503: Adjust Screen for Add Transaction, Edit Transaction Screen

### PR DESCRIPTION
**Pull request by tanbtd-1747**
### Issue Link :link:
https://edu-redmine.sun-asterisk.vn/issues/9503

### Goals :soccer:
- Adjust textfields on screen when keyboard appears, disappears
- Hide keyboard when tapping outside

### CODE REVIEW

**Comment by tanbtd-1747** [comment]
ready

File: Money*/Money*/Scenes/AddTransaction/AddTransactionViewController.swift [hunk 1]
```diff
@@ -28,11 +28,40 @@ final class AddTransactionViewController: UIViewController {
     // MARK: - Properties
     var wallet: Wallet!
     private var transaction = Transaction()
+    var keyboardAdjusted = false
```
Comment (hung070996) [inline Money*/Money*/Scenes/AddTransaction/AddTransactionViewController.swift]: private

File: Money*/Money*/Scenes/AddTransaction/AddTransactionViewController.swift [hunk 2]
```diff
@@ -28,11 +28,40 @@ final class AddTransactionViewController: UIViewController {
...
+    var lastKeyboardOffset: CGFloat = 0
```
Comment (hung070996) [inline Money*/Money*/Scenes/AddTransaction/AddTransactionViewController.swift]: private

File: Money*/Money*/Scenes/AddTransaction/AddTransactionViewController.swift [hunk 3]
```diff
@@ -141,6 +170,35 @@ final class AddTransactionViewController: UIViewController {
     }
 }
 
+// MARK: - Keyboard
+extension AddTransactionViewController {
+    @objc func keyboardWillShow(notification: Notification) {
+        if keyboardAdjusted == false {
+            lastKeyboardOffset = getKeyboardHeight(notification: notification)
+            if nameTextField.isEditing {
```
Comment (hung070996) [inline Money*/Money*/Scenes/AddTransaction/AddTransactionViewController.swift]: lastKeyboardOffset = nameTextField.isEditing ? nameTextField.frame.origin.y : amountTextField.frame.origin.y

File: Money*/Money*/Scenes/AddTransaction/AddTransactionViewController.swift [hunk 4]
```diff
@@ -141,6 +170,35 @@ final class AddTransactionViewController: UIViewController {
...
+    }
+    
+    @objc func keyboardWillHide(notification: Notification) {
+        if keyboardAdjusted == true {
+            view.frame.origin.y += lastKeyboardOffset
+            keyboardAdjusted = false
+        }
+    }
+    
+    func getKeyboardHeight(notification: Notification) -> CGFloat {
```
Comment (hung070996) [inline Money*/Money*/Scenes/AddTransaction/AddTransactionViewController.swift]: private

File: Money*/Money*/Scenes/EditTransaction/EditTransactionViewController.swift [hunk 5]
```diff
@@ -29,14 +29,43 @@ final class EditTransactionViewController: UIViewController {
     var wallet: Wallet!
     var transaction: Transaction!
     var walletBalanceBeforeTransaction: UInt64 = 0
+    var keyboardAdjusted = false
```
Comment (hung070996) [inline Money*/Money*/Scenes/EditTransaction/EditTransactionViewController.swift]: private

File: Money*/Money*/Scenes/EditTransaction/EditTransactionViewController.swift [hunk 6]
```diff
@@ -29,14 +29,43 @@ final class EditTransactionViewController: UIViewController {
...
+    var lastKeyboardOffset: CGFloat = 0
```
Comment (hung070996) [inline Money*/Money*/Scenes/EditTransaction/EditTransactionViewController.swift]: private

File: Money*/Money*/Scenes/EditTransaction/EditTransactionViewController.swift [hunk 7]
```diff
@@ -172,6 +201,35 @@ final class EditTransactionViewController: UIViewController {
...
+    }
+    
+    @objc func keyboardWillHide(notification: Notification) {
+        if keyboardAdjusted == true {
+            view.frame.origin.y += lastKeyboardOffset
+            keyboardAdjusted = false
+        }
+    }
+    
+    func getKeyboardHeight(notification: Notification) -> CGFloat {
```
Comment (hung070996) [inline Money*/Money*/Scenes/EditTransaction/EditTransactionViewController.swift]: private

File: Money*/Money*/Scenes/EditTransaction/EditTransactionViewController.swift [hunk 8]
```diff
@@ -172,6 +201,35 @@ final class EditTransactionViewController: UIViewController {
     }
 }
 
+// MARK: - Keyboard
+extension EditTransactionViewController {
+    @objc func keyboardWillShow(notification: Notification) {
+        if keyboardAdjusted == false {
```
Comment (tienpx3010) [inline Money*/Money*/Scenes/EditTransaction/EditTransactionViewController.swift]: Bỏ keyboardAdjusted

**Comment by tanbtd-1747** [comment]
ready

**Review by hung070996 — APPROVED** [review]

**Review by tienpx3010 — APPROVED** [review]

**Pull request merged**